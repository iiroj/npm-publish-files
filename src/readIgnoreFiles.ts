import * as fs from "fs-extra";

export default async () => {
  const [npmignore, gitignore] = await Promise.all(
    [".npmignore", ".gitignore"].map(
      path =>
        new Promise<Buffer | null>(async resolve => {
          resolve((await fs.pathExists(path)) ? fs.readFile(path) : null);
        })
    )
  );

  if (npmignore || gitignore) {
    // one of the does exist
    return (npmignore || gitignore)!
      .toString()
      .split("\n")
      .filter(Boolean);
  } else {
    return [];
  }
};
