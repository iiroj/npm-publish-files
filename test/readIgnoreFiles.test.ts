import readIgnoreFiles from "../src/readIgnoreFiles";

test("readIgnoreFiles", async () => {
  const ignoreFile = await readIgnoreFiles();
  expect(ignoreFile).toMatchSnapshot();
});
