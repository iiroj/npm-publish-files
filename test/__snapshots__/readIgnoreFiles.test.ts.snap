// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readIgnoreFiles 1`] = `
Array [
  ".rpt2_cache",
  "cli.d.ts",
  "cli.js",
  "coverage",
  "node_modules",
  "test/tmp",
]
`;
